image.c

```diff
@@ -293,6 +300,205 @@ bool img_load_gif(img_t *img, const fileinfo_t *file)
+	/* have only 1 frame) */
+	img->multi.cnt = 0;
+	while (WebPAnimDecoderGetNext(dec, &buf, &ts)) {
+		im = imlib_create_image_using_copied_data(
+				info.canvas_width, info.canvas_height, (DATA32*)buf);
+		imlib_context_set_image(im);
+		imlib_image_set_format("webp");
+		/* Get an iterator of this frame - used for frame info (duration, etc.) */
+		WebPDemuxGetFrame(demux, img->multi.cnt+1, &iter);
+		imlib_image_set_has_alpha((flags & ALPHA_FLAG) > 0);
+		/* Store info for this frame */
+		img->multi.frames[img->multi.cnt].im = im;
+		delay = iter.duration > 0 ? iter.duration : DEF_WEBP_DELAY;
+		img->multi.frames[img->multi.cnt].delay = delay;
+		img->multi.length += img->multi.frames[img->multi.cnt].delay;
+		img->multi.cnt++;
+	}
+	WebPAnimDecoderDelete(dec);
+	WebPDemuxReleaseIterator(&iter);
+
+	if (img->multi.cnt > 1) {
+		imlib_context_set_image(img->im);
+		imlib_free_image();
+		img->im = img->multi.frames[0].im;
+	} else if (img->multi.cnt == 1) {
+		imlib_context_set_image(img->multi.frames[0].im);
+		imlib_free_image();
+		img->multi.cnt = 0;
+	}
+
+	imlib_context_set_image(img->im);
+	imlib_image_set_format("webp");
+
+	free((uint8_t*)data.bytes);
+	return !err;
+}
+
+Imlib_Image load_webp_firstframe_im(const fileinfo_t *file)
+{
+	struct stat st;
+
+	if (!file->name || !file->path)
+		return NULL;
+	if (!STREQ(strrchr(file->path, '.'), ".webp"))
+		return NULL;
+	if (access(file->path, R_OK) != 0 || stat(file->path, &st) != 0 || !S_ISREG(st.st_mode))
+		return NULL;
+
+	FILE *webp_file;
+	size_t file_size;
+	WebPData data;
+
+	Imlib_Image im = NULL;
+	struct WebPAnimDecoderOptions opts;
+	WebPAnimDecoder *dec;
+	struct WebPAnimInfo info;
+	uint8_t *buf = NULL;
+	int ts;
+
+	/* Open the file */
+	if ((webp_file = fopen(file->path, "rb")) == NULL) {
+		error(0, 0, "%s: Error opening webp image", file->name);
+		return NULL;
+	}
+	/* Get the file size */
+	fseek(webp_file, 0L, SEEK_END);
+	file_size = ftell(webp_file);
+	rewind(webp_file);
+	/* Read the whole file into memory */
+	data.bytes = emalloc(file_size);
+	if (fread((uint8_t*)data.bytes, 1, file_size, webp_file) != file_size) {
+		error(0, 0, "%s: Error opening webp image", file->name);
+		free((uint8_t*)data.bytes);
+		return NULL;
+	}
+	data.size = file_size;
+
+	/* Setup the WebP Animation Decoder */
+	if (!WebPAnimDecoderOptionsInit(&opts)) {
+		/* Version mismatch in WebP library */
+		error(0, 0, "%s: Error opening webp image", file->name);
+		free((uint8_t*)data.bytes);
+		return NULL;
+	}
+	opts.color_mode = MODE_BGRA;
+	/* This could cause problems on some systems and will require more testing */
+	/* than I can do (multithreaded decoding) */
+	opts.use_threads = true;
+	if ((dec = WebPAnimDecoderNew(&data, &opts)) == NULL) {
+		/* Parsing error, invalid operation, or memory error */
+		error(0, 0, "%s: Error opening webp image", file->name);
+		free((uint8_t*)data.bytes);
+		return NULL;
+	}
```

Comment:
after some testing i was able to determine that with the previously identified redundant checks removed it is this if statement that will constantly spew out the error message on every non webp image and on extensionless webp images, however simply removing this check will cause a segfault when reading large .jpg images.

Reply:
> large .jpg images.

How large exactly? I'm testing a potential fix which is on the way towards tidying up the patch and I'm not able to reproduce this issue.

Simply removing this check, I get a segfault reading any image with no extension, but it can read a 69MB .jpg image with no problem.

Reply:
i gotta admit that one was quite inconsistent as i could read a 8MB .jpg just fine but a 19MB .jpg just next to it made it segfault, so not truly sure it was exactly that...